sbor/src/schema/well_known_types.rs

```diff
@@ -3,22 +3,87 @@ pub const CUSTOM_WELL_KNOWN_TYPE_START: u8 = 0x80;
     pub const BOOL_ID: u8 = VALUE_KIND_BOOL;
+    pub fn bool_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::Bool)
+    }
+
     pub const I8_ID: u8 = VALUE_KIND_I8;
+    pub fn i8_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::I8)
+    }
+
     pub const I16_ID: u8 = VALUE_KIND_I16;
+    pub fn i16_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::I16)
+    }
+
     pub const I32_ID: u8 = VALUE_KIND_I32;
+    pub fn i32_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::I32)
+    }
+
     pub const I64_ID: u8 = VALUE_KIND_I64;
+    pub fn i64_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::I64)
+    }
+
     pub const I128_ID: u8 = VALUE_KIND_I128;
+    pub fn i128_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::I128)
+    }
+
     pub const U8_ID: u8 = VALUE_KIND_U8;
+    pub fn u8_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::U8)
+    }
+
     pub const U16_ID: u8 = VALUE_KIND_U16;
+    pub fn u16_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::U16)
+    }
+
     pub const U32_ID: u8 = VALUE_KIND_U32;
+    pub fn u32_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::U32)
+    }
+
     pub const U64_ID: u8 = VALUE_KIND_U64;
+    pub fn u64_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::U64)
+    }
+
     pub const U128_ID: u8 = VALUE_KIND_U128;
+    pub fn u128_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::U128)
+    }
+
     pub const STRING_ID: u8 = VALUE_KIND_STRING;
+    pub fn string_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::String)
+    }
+
     pub const ANY_ID: u8 = 0x40; // Any type
+    pub fn any_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::Any)
+    }
+
     pub const BYTES_ID: u8 = 0x41; // `Vec<u8>`
+    pub fn bytes_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::unnamed(TypeKind::Array {
+            element_type: WellKnownTypeIndex(U8_ID).into(),
+        })
+    }
+
     pub const UNIT_ID: u8 = 0x42; // `()`
+    pub fn unit_type_data<C: CustomTypeKind<L>, L: SchemaTypeLink>() -> TypeData<C, L> {
+        TypeData::no_child_names(
+            TypeKind::Tuple {
+                field_types: sbor::rust::prelude::vec![],
+            },
+            "None",
```

Comment:
Is `None` preferred? I personally like `Unit` more, following Rust's terminologies.

Reply:
So when I originally envisioned this, the name was going to be shown in (eg) wallets/dashboards etc, in the user display of types. Where Unit would be out of place.

But actually I think I should leave this empty and leave it up to the UI to decide how to display this - so will make it unnamed.